### DISCUSSION TITLE
loading state/spinner for ref.watch inside future provider

### DISCUSSION THREAD

**Discussion by neiljaywarner** [rrousselGit/riverpod#1600] [Q&A] [answered]
Can i get some help with this please?  I thought this might be nice to put here in case there's an area for improvement in documentation or it is otherwise helpful to someone.

https://stackoverflow.com/questions/73588646/how-do-i-get-loading-state-spinner-with-flutter-riverpod-statenotifier-ref-watc
When i change paramProvider I would expect it to get a loading state if it is going to successfully get a new data state.
instead it gets a new data state without a new loading state.  Sorry if this is a dumb question.
NOTE: Someone on stackoverflow says it works with  Flutter 3.3.0 And using hooks_riverpod: ^1.0.4
If this is correct and then i try recent riverpod i guess i will file bug report. 

`final paramProvider = StateProvider((ref) => 'Initial value');

final futureProvider = FutureProvider.autoDispose<String>((ref) async {
  String parameter = ref.watch(paramProvider);
  return pretendApiCall(parameter);
});
Future<String> pretendApiCall(String parameter) async {
  await Future.delayed(const Duration(seconds: 3));
  return parameter;
}`

**Comment by rrousselGit** [comment, marked as answer]
This behavior has changed on the master branch

Simply wait for a release and your problem should disappear

**Reply by neiljaywarner** [reply]
@rrousselGit that's cool.  I'm just curious -for now then will the functionality of keeping the old value be gone? I would think that the spinner is better - it's confusing if the user taps a button and sees no result at all.

**Reply by mcrio** [reply]
Adding additional info for those of us landing here and asking the same question:

[Changelog](https://pub.dev/packages/riverpod/changelog) 
> 2.0.0-dev.0
> 
> Breaking After a provider has emitted an AsyncValue.data or AsyncValue.error, that provider will no longer emit an AsyncValue.loading. Instead, it will re-emit the latest value, but with the property AsyncValue.isRefreshing to true.
> 
> This allows the UI to keep showing the previous data/error when a provider is being refreshed.

**Comment by neiljaywarner** [comment]
Hm. I could  probably close this except i didn't get it workign with riverpod 2.x in my actual project.  still, hopefully this is a good discussion. the answer in the stackoverflow question
`asyncValue.isRefreshing`
